### ISSUE TITLE
Feature request: add attribute to upcloud_gateway to get public address of gateway

### ISSUE THREAD

**Issue by pawwe** [UpCloudLtd/terraform-provider-upcloud#411]
When creating a NAT gateway, it would be nice to have the public address attribute exposed.

There is no attribute I can use to get the IPv4 address assigned to the gateway. I need to use the [`http`](https://registry.terraform.io/providers/hashicorp/http) provider to get it using the API like this:

```terraform
resource "upcloud_gateway" "this" {
  name     = var.gateway.name
  zone     = var.zone.name
  features = var.gateway.features
  router {
    id = upcloud_router.this.id
  }
}

data "http" "this" {
  url = "https://api.upcloud.com/1.3/gateway/${upcloud_gateway.this.id}"
  request_headers = {
    Accept        = "application/json"
    Authorization = "Basic ${base64encode("${var.UPCLOUD_USERNAME}:${var.UPCLOUD_PASSWORD}")}"
  }
}

output "this" {
  value = jsondecode(data.http.this.response_body).addresses[0].address
}
```

Also, the [API documentation for network gateways](https://developers.upcloud.com/1.3/19-network-gateways/#get-service-details) needs to be updated. It doesn't mention anywhere that the address can be found with the API.

```
Terraform v1.6.0
on linux_amd64
+ provider registry.terraform.io/upcloudltd/upcloud v2.12.0
```

**Comment by kangasta** [comment]
Hello, thanks for the feature request! This is already implemented, but [pending release](https://github.com/UpCloudLtd/terraform-provider-upcloud/blob/main/CHANGELOG.md#added). We will try to create a new release during this week to make this available.

**Comment by kangasta** [comment]
Sorry for the delay! We are adding few more features to the release. The new release should be available early next week.

**Comment by kangasta** [comment]
This is now available in [v3.0.0](https://github.com/UpCloudLtd/terraform-provider-upcloud/releases/tag/v3.0.0)

**Comment by kangasta** [comment]
I'll close this as the feature has been released. Please re-open or create a bug ticket if there are problems with the feature.
